protocols/kad/src/query/peers/closest/disjoint.rs

```diff
@@ -447,21 +447,21 @@ mod tests {
     use crate::K_VALUE;
     use libp2p_core::multihash::{Code, Multihash};
     use quickcheck::*;
-    use rand::{seq::SliceRandom, Rng};
     use std::collections::HashSet;
     use std::iter;
 
     impl Arbitrary for ResultIter<std::vec::IntoIter<Key<PeerId>>> {
-        fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        fn arbitrary(g: &mut Gen) -> Self {
             let target = Target::arbitrary(g).0;
-            let num_closest_iters = g.gen_range(0, 20 + 1);
-            let peers = random_peers(g.gen_range(0, 20 * num_closest_iters + 1), g);
+            let num_closest_iters = g.gen_range(0..20 + 1);
+            let peers = random_peers(g.gen_range(0..20 * num_closest_iters + 1), g);
 
             let iters: Vec<_> = (0..num_closest_iters)
                 .map(|_| {
-                    let num_peers = g.gen_range(0, 20 + 1);
-                    let mut peers = peers
-                        .choose_multiple(g, num_peers)
+                    let num_peers = g.gen_range(0..20 + 1);
+                    let mut peers = g
+                        .choose_multiple(&peers, num_peers)
+                        .into_iter()
```

Comment:
In `choose_multiple` we already had an iterator. Perhaps we can expose that iterator directly instead of `collect`ing first and then making an iterator again?